### DISCUSSION TITLE
Rewind

### DISCUSSION THREAD

**Discussion by MarcFinns** [earlephilhower/ESP8266Audio#708] [Q&A]
hi there, is there a proper way to play again a file, or the only approach is to delete the file/generator objects and instantiate them again?
thanks!

**Comment by dbu** [comment]
i tried to use the `seek` call, but it did not work for me (not sure if i did something wrong or if the file gets closed when reaching the end).

i did not investigate long, because looking at https://github.com/earlephilhower/ESP8266Audio/blob/master/src/AudioFileSourceFS.h made me realize i don't need an instance of the source for each of my files but can instead use the `close()` and `open("filename")` calls on a single the audio file source instance, which made things even easier in my use case.
